### ISSUE TITLE
Crash with Biomes O’ Plenty on loading dimension (“Climate lookup failed”)

### ISSUE THREAD

**Issue by Dimentive** [Ivorforce/RecurrentComplex#386]
I teleported from the Nether to the overworld using RFTools’ advanced charged porter, and my game immediately crashed. I had already teleported back and forth using the advanced charged porter many times, so I don’t know why I only crashed once. The crash seems to be related to both Recurrent Complex and [Biomes O’ Plenty](https://minecraft.curseforge.com/projects/biomes-o-plenty).

[Crash report.](https://pastebin.com/raw/a6KqkT5y)

#### Versions

* **Modpack:** [Enigmatica 2](https://minecraft.curseforge.com/projects/enigmatica2) (version [1.63a](https://minecraft.curseforge.com/projects/enigmatica2/files/2691579), latest)
* **Minecraft:** 1.12.2
* **Forge:** 14.23.5.2781
* **Recurrent Complex:** [1.4.8.2](https://minecraft.curseforge.com/projects/recurrent-complex/files/2689602) (latest)
* **IvToolkit:** [1.1.3](https://minecraft.curseforge.com/projects/ivtoolkit/files/2443253) (latest)

**Comment by Ivorforce** [comment]
We've had this before, didn't we? I thought I saw this earlier.
Anyway, this is a BoP crash. It shouldn't have anything to do with Recurrent Complex - please report to the respective authors.

**Comment by Dimentive** [comment]
No, I haven’t gotten or reported this crash before. [I did mention another crash](https://github.com/Ivorforce/RecurrentComplex/issues/374#issuecomment-474334651) that was also related to Recurrent Complex and Biomes O’ Plenty last month, but that was not the same kind of crash. That crash was a `java.lang.OutOfMemoryError: Java heap space` crash, while this crash is a `java.lang.RuntimeException: Climate lookup failed climateOrdinal` crash.

The reason I reported it here was that Recurrent Complex was mentioned in the crash report.

    at ivorius.reccomplex.world.gen.feature.structure.MapGenStructureHook.generate(MapGenStructureHook.java:153)
    at ivorius.reccomplex.events.handlers.RCTerrainGenEventHandler$3.generate(RCTerrainGenEventHandler.java:245)

I just noticed that [OptiFine](https://optifine.net) was mentioned too. I don’t know whether OptiFine has anything to do with this crash. It could have.

I’m not sure which mod actually caused the crash; it could be a conflict between RC and BOP or a conflict with OptiFine and the two mods. I, unfortunately, am not allowed to report this on Biomes O’ Plenty’s issue tracker, however, as the BOP team will not fix bugs or provide support if you have OptiFine installed.
